src/promql/src/extension_plan/instant_manipulate.rs

```diff
@@ -127,6 +135,40 @@ impl InstantManipulate {
             metric: ExecutionPlanMetricsSet::new(),
         })
     }
+
+    pub fn serialize(&self) -> Vec<u8> {
+        pb::InstantManipulate {
+            start: self.start,
+            end: self.end,
+            interval: self.interval,
+            lookback_delta: self.lookback_delta,
+            time_index: self.time_index_column.clone(),
+            field_index: self.field_column.clone().unwrap_or_default(),
+        }
+        .encode_to_vec()
+    }
+
+    pub fn deserialize(bytes: &[u8]) -> Result<Self> {
+        let pb_instant_manipulate =
+            pb::InstantManipulate::decode(bytes).context(DeserializeSnafu)?;
+        let placeholder_plan = LogicalPlan::EmptyRelation(EmptyRelation {
+            produce_one_row: false,
+            schema: Arc::new(DFSchema::empty()),
+        });
+        Ok(Self::new(
+            pb_instant_manipulate.start,
+            pb_instant_manipulate.end,
+            pb_instant_manipulate.lookback_delta,
+            pb_instant_manipulate.interval,
+            pb_instant_manipulate.time_index,
+            if pb_instant_manipulate.field_index.is_empty() {
+                None
+            } else {
+                Some(pb_instant_manipulate.field_index)
+            },
+            placeholder_plan,
```

Comment:
I prefer to use struct assignment, the `new` function has many parameters and it is easy to get the order wrong.